cmd/kpromo/cmd/pr/pr.go

```diff
@@ -0,0 +1,441 @@
+		}
+	}()
+
+	// Path to the promoter image list
+	imagesListPath := filepath.Join(
+		image.ProdRegistry,
+		"images",
+		filepath.Base(image.StagingRepoPrefix)+opts.project,
+		"images.yaml",
+	)
+
+	// Read the current manifest to check later if new images come up
+	oldlist := make([]byte, 0)
+
+	// Run the promoter manifest grower
+	if mustRun(opts, "Grow the manifests to add the new tags?") {
+		if util.Exists(filepath.Join(repo.Dir(), imagesListPath)) {
+			logrus.Debug("Reading the current image promoter manifest (image list)")
+			oldlist, err = os.ReadFile(filepath.Join(repo.Dir(), imagesListPath))
+			if err != nil {
+				return errors.Wrap(err, "while reading the current promoter image list")
+			}
+		}
+
+		for _, tag := range opts.tags {
+			opt := reg.GrowManifestOptions{}
+			if err := opt.Populate(
+				filepath.Join(repo.Dir(), image.ProdRegistry),
+				image.StagingRepoPrefix+opts.project, "", "", tag); err != nil {
+				return errors.Wrapf(err, "populating image promoter options for tag %s", tag)
+			}
+
+			if err := opt.Validate(); err != nil {
+				return errors.Wrapf(err, "validate promoter options for tag %s", tag)
+			}
+
+			logrus.Infof("Growing manifests with images matching tag %s", tag)
+			if err := reg.GrowManifest(ctx, &opt); err != nil {
+				return errors.Wrapf(err, "Growing manifest with tag %s", tag)
+			}
+		}
+	}
+
+	// Re-write the image list without the mock images
+	rawImageList, err := image.NewManifestListFromFile(filepath.Join(repo.Dir(), imagesListPath))
+	if err != nil {
+		return errors.Wrap(err, "parsing the current manifest")
+	}
+
+	// Create a new imagelist to copy the non-mock images
+	newImageList := &image.ManifestList{}
+
+	// Copy all non mock-images:
+	for _, imageData := range *rawImageList {
+		if !strings.Contains(imageData.Name, "mock/") {
+			*newImageList = append(*newImageList, imageData)
+		}
+	}
+
+	// Write the modified manifest
+	if err := newImageList.Write(filepath.Join(repo.Dir(), imagesListPath)); err != nil {
+		return errors.Wrap(err, "while writing the promoter image list")
+	}
+
+	// Check if the image list was modified
+	if len(oldlist) > 0 {
+		logrus.Debug("Checking if the image list was modified")
+		// read the newly modified manifest
+		newlist, err := os.ReadFile(filepath.Join(repo.Dir(), imagesListPath))
+		if err != nil {
+			return errors.Wrap(err, "while reading the modified manifest images list")
+		}
+
+		// If the manifest was not modified, exit now
+		if string(newlist) == string(oldlist) {
+			logrus.Info("No changes detected in the promoter images list, exiting without changes")
+			return nil
+		}
+	}
+
+	// add the modified manifest to staging
+	logrus.Debugf("Adding %s to staging area", imagesListPath)
+	if err := repo.Add(imagesListPath); err != nil {
+		return errors.Wrap(err, "adding image manifest to staging area")
+	}
+
+	commitMessage := "releng: Image promotion for " + opts.project + " " + strings.Join(opts.tags, " / ")
```

Comment:
small nit I had from using this recently: do you think we could get rid of the `releng:` prefix? or maybe just add it when it's for the Kubernetes project? Not sure if it's relevant for other projects

Reply:
Mind if I fix in a follow-up?
There are more things I'd like to fix about the message, but they're a little out-of-scope for this PR.

Reply:
sounds good 👌

Reply:
> small nit I had from using this recently: do you think we could get rid of the `releng:` prefix? or maybe just add it when it's for the Kubernetes project? Not sure if it's relevant for other projects

Will be fixed in https://github.com/kubernetes-sigs/promo-tools/pull/460.